src/middleware/auto-webp.ts

```diff
@@ -0,0 +1,150 @@
+import { proxy } from "../proxy";
+import { Image } from "@fly/v8env/lib/fly/image/index";
+import { get } from "@fly/v8env/lib/fly/cache/response";
+import { set } from "@fly/v8env/lib/fly/cache/response";
+
+/**
+ * Automatically encodes images as webp for supported clients.
+ * This would apply to requests with 'image/webp' in the accept header 
+ * and 'image/jpeg' or 'image/png' in the response type from origin.
+ * 
+ * Example:
+ * 
+ * ```typescript
+ * import { imageService } from "./src/middleware/auto-webp";
+ *
+ * const images = imageService(
+ *   "http://www.example.com/"
+ * )
+ *
+ * fly.http.respondWith(images)
+ * ```
+ */
+
+export interface ImageServiceOptions {
+  rootPath?: string,
+  webp?: boolean
+}
+
+export type FetchFn = (req: RequestInfo, init?: RequestInit) => Promise<Response>
+
+export function imageService(origin: FetchFn | string): FetchFn {
```

Comment:
These function names need changing to match the file/import name. Maybe something like `autoWebP`?

Also this function only needs to take a `FetchFunction`. I'm fairly sure `FetchFn` doesn't exist in this particular project, and a string will never get used in this context.